statsmodels/distributions/empirical_distribution.py

```diff
@@ -132,15 +132,13 @@ class ECDF(StepFunction):
     array([ 0.75,  1.  ,  0.  ,  0.25])
     """
     def __init__(self, x, side='right'):
-        step = True
```

Comment:
This is a lossy change since it loses what to do it step is added as an option. Could leave as a well-formatted comment.

Reply:
The other branch hides a NameError since `ival` isn't defined (besides which it has a `return` in it).  So the alternative is to replace line 143 with raising NotImplementedError.  I prefer this route because it isn't hard-coding a branching.

Reply:
I’m open to ideas if there’s a third option

Reply:
Something like
```
# TODO: make step an arg and have a linear interpolation option?
# This is the path for step is True
# If step is False, interp1d(x,y,drop_errors=False,fill_values=ival)
# which currently raises a NameError, and so would need to be fixed
```

Reply:
Made the suggested change.

This is one of the occasions where I'd be on board with the "thats what git history is for" approach to keeping the old version accessible (or just a reference back to this thread).

Reply:
De facto green

Reply:
Thanks.  These types of comments would help should someone every try and rescue this code. They make it much easier to merge changes like this.